cmd/atlas/internal/cloudapi/client.go

```diff
@@ -0,0 +1,85 @@
+package cloudapi
+
+import (
+	"context"
+	"encoding/base64"
+	"net/http"
+	"time"
+
+	"ariga.io/atlas/cmd/atlas/internal/cloudapi/graphql"
+	"ariga.io/atlas/sql/migrate"
+)
+
+const (
+	UserAgent = "atlas-cli"
+)
+
+// Client is a client for the Atlas Cloud API.
+type Client struct {
+	client   graphql.Client
+	endpoint string
+}
+
+// New creates a new Client for the Atlas Cloud API.
+func New(endpoint, token string) *Client {
+	return &Client{
+		client: graphql.NewClient(endpoint, &http.Client{
+			Transport: &roundTripper{
+				token: token,
+			},
+			Timeout: time.Second * 30,
+		}),
+	}
+}
+
+// GetDir retrieves a directory from the Atlas Cloud API.
+func (c *Client) GetDir(ctx context.Context, input DirInput) (migrate.Dir, error) {
+	var payload struct {
+		Dir struct {
+			Content string `json:"content"`
```

Comment:
```suggestion
			Content []byte `json:"content"`
```
If the field is []byte, json.Unamarshal will handle the base64 phase.